aiohttp/test_utils.py

```diff
@@ -42,27 +36,26 @@ def __init__(self, *, scheme=sentinel, loop=None,
                  host='127.0.0.1', port=None, skip_url_asserts=False,
                  **kwargs):
         self._loop = loop
-        self.port = None
-        self.server = None
-        self.handler = None
+        self.runner = None
         self._root = None
         self.host = host
+        self.port = port
         self._closed = False
         self.scheme = scheme
         self.skip_url_asserts = skip_url_asserts
-        self.port = port
 
     async def start_server(self, loop=None, **kwargs):
-        if self.server:
+        if self.runner:
             return
         self._loop = loop
-        self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        if self.port:
-            self._socket.bind((self.host, self.port))
-        else:
-            self._socket.bind((self.host, 0))
-            self.port = self._socket.getsockname()[1]
         self._ssl = kwargs.pop('ssl', None)
+        self.runner = await self._make_runner(**kwargs)
+        await self.runner.setup()
+        if not self.port:
+            self.port = unused_port()
+        site = TCPSite(self.runner, host=self.host, port=self.port,
```

Comment:
Previously, if port was None, it turns into 0 e.g. random. Now it would set as some default `8443 if self._ssl_context else 8080`. Is that ok?

Reply:
Also, what about port/address reuse? Sounds like they are pretty useful in tests where server starts and shuts down quite frequently.

Reply:
Ah, nvm about zero port - missed `unused_port()` call right in front of me /:

Reply:
asyncio reuses address on UNIX, should be enough.

Reply:
Good to know!